lib/ansible/template/__init__.py

```diff
@@ -510,10 +531,13 @@ def __getitem__(self, key):
                 for func_name, func in iteritems(method_map()):
                     fq_name = '.'.join((parent_prefix, func_name))
                     # FIXME: detect/warn on intra-collection function name collisions
-                    if self._jinja2_native and func_name in C.STRING_TYPE_FILTERS:
-                        self._collection_jinja_func_cache[fq_name] = _wrap_native_text(func)
+                    if self._pluginloader.class_name == 'FilterModule':
```

Comment:
I think the original code (and thus this code as well) might be incorrect here- we should only be applying this for builtin filters that match `STRING_TYPE_FILTERS`, not any collection-hosted filters that happen to have the same name...

Reply:
Good catch. Since `STRING_TYPE_FILTERS` is configurable we should also allow for collection-hosted filters to be specified there as FQCN, so something like the below?
```diff
diff --git a/lib/ansible/template/__init__.py b/lib/ansible/template/__init__.py
index bfc0251408..c37ab3f9f7 100644
--- a/lib/ansible/template/__init__.py
+++ b/lib/ansible/template/__init__.py
@@ -510,7 +510,7 @@ class JinjaPluginIntercept(MutableMapping):
                 for func_name, func in iteritems(method_map()):
                     fq_name = '.'.join((parent_prefix, func_name))
                     # FIXME: detect/warn on intra-collection function name collisions
-                    if self._jinja2_native and func_name in C.STRING_TYPE_FILTERS:
+                    if self._jinja2_native and (fq_name in C.STRING_TYPE_FILTERS or (fq_name.startswith.(('ansible.builtin.', 'ansible.legacy')) and func_name in C.STRING_TYPE_FILTERS)):
                         self._collection_jinja_func_cache[fq_name] = _wrap_native_text(func)
                     else:
                         self._collection_jinja_func_cache[fq_name] = _unroll_iterator(func)
```

Either way this should be fixed in a separate PR so it is easier to backport to `2.10`.

Reply:
Yeah, separate PR is fine, though we should also not be relying on config for that for anything but the builtins and maybe local non-collection plugins- collections should be able to provide that metadata directly (probably via something like the metadata-only decorator stuff I was playing with).

Reply:
Opened https://github.com/ansible/ansible/pull/71801 to address this.

Reply:
Fixed in https://github.com/ansible/ansible/pull/71801